src/main/java/crawlercommons/sitemaps/SiteMapTester.java

```diff
@@ -32,37 +32,50 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(SiteMapTester.class);
     private static SiteMapParser parser = new SiteMapParser(false);
+    private static SiteMapParser saxParser = new SiteMapParserSAX(false, true);
 
     public static void main(String[] args) throws IOException, UnknownFormatException {
         if (args.length < 1) {
-            LOG.error("Usage: SiteMapTester <URL_TO_TEST> [MIME_TYPE]");
+            LOG.error("Usage: SiteMapTester [--use-sax-parser] <URL_TO_TEST> [MIME_TYPE]");
         } else {
-            URL url = new URL(args[0]);
-            String mt = (args.length > 1) ? args[1] : null;
+            int i = 0;
+            boolean useSaxParser = false;
+            if ("--use-sax-parser".equals(args[i])) {
+                useSaxParser = true;
+                i++;
+            }
+            URL url = new URL(args[i]);
+            i++;
+            String mt = (args.length > i) ? args[i] : null;
 
-            parse(url, mt);
+            parse(url, mt, useSaxParser);
         }
     }
 
     /**
      * Parses a Sitemap recursively meaning that if the sitemap is a
      * sitemapIndex then it parses all of the internal sitemaps
      */
-    private static void parse(URL url, String mt) throws IOException, UnknownFormatException {
+    private static void parse(URL url, String mt, boolean useSaxParser) throws IOException, UnknownFormatException {
```

Comment:
Ok, you're right, removing arguments can break things. I'll update the PR.
But I wouldn't call SiteMapTester an API (the "tester" indicates it): it's not thread-safe due to the usage of a static parser object. If we want to keep a stable API after removing the DOM-based parser and rename SiteMapParserSAX to SiteMapParser, a factory which is configured by properties, could be the better way.